### ISSUE TITLE
[New Language] Transpiled Go Lang port of CCXT

### ISSUE THREAD

**Issue by DavidXanatos** [ccxt/ccxt#9934]
We have created a transpiled version of CCXT for Go that’s working with 100+ exchanges: https://github.com/prompt-cash/ccxt-go 
 
It’s all still very alpha, so no guarantees of any kind. Especially some rarely used helper functions are not implemented yet and just panic with a todo message. If you need these exchanges you will need to implement those.
 
We are aware others tried to do this before, but it looks like they never finished the work to actually transpile and run the code of all exchanges: https://github.com/ccxt/ccxt/pull/3651 
 
If you are writing code in Go, we would welcome some feedback. Additionally we would appreciate it if the CCXT maintainers can outline a path on how to officially integrate this into CCXT.

**Comment by kroitor** [comment]
Hey!

First of all, thanks for your involvement! Great job!

> Additionally we would appreciate it if the CCXT maintainers can outline a path on how to officially integrate this into CCXT.

It starts with automating the transpilation process within the build sequence (this is done on Travis CI for CCXT: https://app.travis-ci.com/github/ccxt/ccxt/). I took a quick overview of the ccxt-go repository. As far as I see, the repository contains the result of transpilation, but I could not find the transpiling code itself, can you please point me to it, or paste a link here? I am not very familiar with Go, but I have a C/C++ background, and we we will be happy to add an extra build step for Go in CCXT master with your help.

For direct communication and coordination you can also ping us on our discord chat:

[![Discord](https://img.shields.io/discord/690203284119617602?logo=discord&logoColor=white)](https://discord.gg/dhzSKYU)

Please, don't hesitate if anything.

**Comment by Darkbladecr** [comment]
Exciting stuff @DavidXanatos would be nice to see how you have built the transpiration process.

Will have a play around when I get some time later this week.

**Comment by zsyh** [comment]
It's a milestone!  It means transpiling to C++ is also possible.
I see the transpiler is at github.com/prompt-cash/js-transpiler, not yet open sourced.

**Comment by DavidXanatos** [comment]
> 
> 
> It's a milestone! It means transpiling to C++ is also possible.
> I see the transpiler is at https://github.com/prompt-cash/ccxt-go/blob/master/pkg/transpiler/transpiler.go

Yes in fact a good portion of the transpiler I had initially written in C/C++ for an other project, and I have also a CVariant class that could be used here with C/C++ so no need to write a new one like in GO.

**Comment by acaloiaro** [comment]
@DavidXanatos Thank you for your efforts!  
@kroitor I'm happy to help get this PR to a state where the project may consider it mergeable if you can provide a rough description of what's required for Go to gain first class support.

**Comment by kroitor** [comment]
@acaloiaro thanks for letting me know, pardon for the delay with my reply, I'm a bit overloaded rn, but I will do my best to post my thoughts on it in the coming days.

**Comment by DOBss** [comment]
any news on this issue? would be really nice to be able to port our implementation to golang

**Comment by ttodua** [comment]
as historical reference, here should be also the link to [previous topic](https://github.com/ccxt/ccxt/issues/1952) about Go implementation.

**Comment by comalice** [comment]
We're two years out from this (and five years out from the previous topic). What needs to happen to get this into shape to merge into CCXT main? I'd be willing to shoulder some of the work to get this done as long as I can get some guidance on what needs to be done.

**Comment by kroitor** [comment]
@comalice thanks, we will try to dedicate some of our human resources to this issue in order to move forward with the support of Go, hopefully soon!

**Comment by arki-joon-010** [comment]
we love golang

**Comment by ttodua** [comment]
closing this issue, as CCXT implemented GO lang in the recent past.
